### PR TITLE
Address TypeError in extract_table.py.  Resolves #76

### LINKED ISSUE #76: TypeError in extract_table.py
**Describe the bug**
It appears that in TextractTableExtractor._extract(), in the processing of tables, element.text_representation is None.

**To Reproduce**
Do something like this with Sycamore:
```
ds = ctx.read.binary(paths,
                     binary_format='pdf',
                     filesystem=s3) \
       .partition(partitioner=UnstructuredPdfPartitioner(),
                  table_extractor=TextractTableExtractor(
                      region_name='us-east-1')) \
```
...and get this...
```
Traceback (most recent call last):
  File "/home/alex/github/sycamore/foobar.py", line 64, in <module>
    ds.show()
  File "/home/alex/github/sycamore/sycamore/docset.py", line 48, in show
    documents = [Document(row) for row in dataset.take(limit)]
                                          ^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/dataset.py", line 2185, in take
    for row in self.iter_rows():
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/iterator.py", line 237, in iter_rows
    for batch in self.iter_batches(**iter_batch_args):
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/iterator.py", line 159, in iter_batches
    block_iterator, stats, blocks_owned_by_consumer = self._to_block_iterator()
                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/iterator/iterator_impl.py", line 32, in _to_block_iterator
    block_iterator, stats, executor = ds._plan.execute_to_iterator()
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/plan.py", line 538, in execute_to_iterator
    block_iter = itertools.chain([next(gen)], gen)
                                  ^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/legacy_compat.py", line 51, in execute_to_legacy_block_iterator
    for bundle in bundle_iter:
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/interfaces.py", line 548, in __next__
    return self.get_next()
           ^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/streaming_executor.py", line 129, in get_next
    raise item
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/streaming_executor.py", line 187, in run
    while self._scheduling_loop_step(self._topology) and not self._shutdown:
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/streaming_executor.py", line 235, in _scheduling_loop_step
    process_completed_tasks(topology)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/streaming_executor_state.py", line 333, in process_completed_tasks
    op.notify_work_completed(ref)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/operators/task_pool_map_operator.py", line 65, in notify_work_completed
    task.output = self._map_ref_to_ref_bundle(ref)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/operators/map_operator.py", line 357, in _map_ref_to_ref_bundle
    all_refs = list(ray.get(ref))
                    ^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/_private/auto_init_hook.py", line 24, in auto_init_wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/_private/worker.py", line 2524, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(TypeError): ray::Map(prepend_scheme)->Map(BinaryScan._to_document)->Map(ray_callable)->Map(ray_callable)->Map(ray_callable)->FlatMap(ray_callable)() (pid=89488, ip=10.1.10.95)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/execution/operators/map_operator.py", line 415, in _map_task
    for b_out in fn(iter(blocks), ctx):
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/plan_udf_map_op.py", line 76, in do_map
    yield from transform_fn(blocks, ctx, *fn_args, **fn_kwargs)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/flat_map.py", line 23, in fn
    for block in blocks:
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/plan_udf_map_op.py", line 76, in do_map
    yield from transform_fn(blocks, ctx, *fn_args, **fn_kwargs)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/map_rows.py", line 23, in fn
    for block in blocks:
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/plan_udf_map_op.py", line 76, in do_map
    yield from transform_fn(blocks, ctx, *fn_args, **fn_kwargs)
  File "/home/alex/.cache/pypoetry/virtualenvs/sycamore-ai--WPxFaDf-py3.11/lib/python3.11/site-packages/ray/data/_internal/planner/map_rows.py", line 26, in fn
    item = row_fn(row)
           ^^^^^^^^^^^
  File "/home/alex/github/sycamore/sycamore/transforms/map.py", line 13, in ray_callable
    document = f(Document(input_dict))
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/github/sycamore/sycamore/transforms/extract_table.py", line 55, in extract_tables
    tables = self._extract(document)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/alex/github/sycamore/sycamore/transforms/extract_table.py", line 43, in _extract
    element.text_representation = element.text_representation + table.to_csv() + "\n"
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

**Expected behavior**
No error.

**Desktop (please complete the following information):**
 - OS: Linux

**Additional context**
I can provide documents that trigger the problem privately.

### PR DESCRIPTION
**Pull request by alexaryn**

### CODE REVIEW

File: sycamore/transforms/partition.py
Comment (bohou-aryn) [inline sycamore/transforms/partition.py:1]: might need have a separate branch for this pr.

**Comment by alexaryn** [comment]
Somehow got an extra commit in here.  Abandoning.

File: sycamore/transforms/extract_table.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def bbox_to_coord(bbox):
             if table.title:
                 element.text_representation = table.title.text + "\n"
                 element.properties["boxes"].append(bbox_to_coord(table.title.bbox))
+            else:
+                element.text_representation = ""
```
Comment (bohou-aryn) [inline sycamore/transforms/extract_table.py:43]: yes, as you mentioned in issue, it's also worthwhile to initialize to empty string.

Comment (alexaryn) [reply sycamore/transforms/extract_table.py:43]: I'm happy to initialize to empty string, but I worry that some code may depend on the difference between None and "" for some logic.  It's a bigger, more general change.

**Pull request closed without merging**